src/commands/lint/mod.rs

```diff
@@ -1,35 +1,48 @@
 pub use args::*;
-#[cfg(feature = "rustup")]
-use serde::Deserialize;
+use tracing::error;
 
 #[cfg(feature = "rustup")]
-use crate::external_cli::Package;
+use crate::commands::lint::install::install_linter;
+use crate::{
+    bin_target::select_run_binary,
+    commands::lint::install::list,
+    config::CliConfig,
+    external_cli::cargo::{
+        self,
+        install::{AutoInstall, is_installed},
+    },
+};
 
 mod args;
-
-/// Represents the contents of `rust-toolchain.toml`.
-#[cfg(feature = "rustup")]
-#[derive(Deserialize)]
-struct RustToolchain {
-    toolchain: Toolchain,
-}
-
-#[cfg(feature = "rustup")]
-#[derive(Deserialize)]
-struct Toolchain {
-    channel: String,
-}
+mod install;
 
 /// Runs `bevy_lint`, if it is installed, with the given arguments.
 ///
 /// Calling `lint(vec!["--workspace"])` is equivalent to calling `bevy_lint --workspace` in the
 /// terminal.
 pub fn lint(args: &mut LintArgs) -> anyhow::Result<()> {
+    const PROGRAM: &str = "bevy_lint";
     use anyhow::ensure;
 
-    #[cfg(feature = "web")]
-    use crate::commands::lint::args::LintSubcommands;
-    use crate::{bin_target::select_run_binary, config::CliConfig, external_cli::cargo};
+    #[cfg(feature = "rustup")]
+    if let Some(LintSubcommands::List) = args.subcommand {
+        return list();
+    } else if let Some(LintSubcommands::Install(args)) = &args.subcommand {
+        return install_linter(args);
+    }
+
+    #[cfg(not(feature = "rustup"))]
+    if let Some(LintSubcommands::List) = args.subcommand {
+        return list();
+    }
```

Comment:
```suggestion
    if let Some(LintSubcommands::List) = args.subcommand {
        return list();
    }

    #[cfg(feature = "rustup")]
    if let Some(LintSubcommands::Install(args)) = &args.subcommand {
        return install_linter(args);
    }
```

Having two different checks for the `list` subcommand is confusing, this detangles the branches so it's easier to figure out what's happening :)